robottelo/performance/graph.py

```diff
@@ -0,0 +1,97 @@
+import pygal
+
+
+def generate_bar_chart_stat(stat_dict, head, filename, lengend):
+    """Generate Bar chart for stat on concurrent subscription"""
+    bar_chart = pygal.Bar()
+    bar_chart.title = head
+    bar_chart.x_labels = ('min', 'mean', 'max', 'std')
+    bar_chart.x_title = 'Statistics'
+    bar_chart.y_title = 'Time (s)'
+    for key in range(len(stat_dict)):
+        bar_chart.add(
+            '{0}-{1}'.format(lengend, key),
+            stat_dict.get(key)
+        )
+    bar_chart.render_to_file(filename)
+
+
+def generate_line_chart_bucketized_stat(stat_dict, head, filename,
+                                        bucket_size, num_buckets):
+    """Generate Line chart for stat on concurrent subscription"""
+    if bucket_size == 0:
+        return
+
+    # parameters for formats on charts
+    lengend = ('min', 'median', 'max', 'std')
+    buckets = [
+        '{0}-{1}'.format(
+            bucket_size * i + 1,
+            bucket_size * (i + 1)
+        )
+        for i in range(num_buckets)
+    ]
+    line_chart = pygal.Line()
+    line_chart.title = head
+    line_chart.x_labels = buckets
+    line_chart.x_title = 'Buckets'
+    line_chart.y_title = 'Time (s)'
+    for stat, stat_value in enumerate(lengend):
+        # count min/median/max/std separately
+        tmp_list = []
+        # accumulate stat item of each client
+        for index in range(len(stat_dict)):
+            tmp_list.append(stat_dict.get(index)[stat])
+        # add list to chart
+        line_chart.add(
+            lengend[stat],
+            tmp_list
+        )
+    line_chart.render_to_file(filename)
+
+
+def generate_stacked_line_chart_raw(time_result_dict, head, filename):
+    """Generate Stacked-Line chart for raw data of ak/att/del/reg"""
+    stackedline_chart = pygal.StackedLine(
+        fill=True, show_dots=False, range=(0, 50)
+    )
+    # compute labels
+    max_label = len(time_result_dict.get('thread-0'))
+    stackedline_chart.x_labels = map(str, range(1, max_label))
```

Comment:
Please use a comprehension of some sort instead of `map`. `map` behaves differently on Python 2 vs Python 3, `map` is less powerful than a comprehension, and it's better to force readers to learn one mechanism (comprehensions) than two or three (map/filter/lambda).

You most likely want to use the list comprehension instead of the generator, as `map` produces a list under Python 2:

```python
>>> [str(i) for i in range(1, 10)]
['1', '2', '3', '4', '5', '6', '7', '8', '9']
>>> (str(i) for i in range(1, 10))
<generator object <genexpr> at 0x7ff532462f30>
```